packages/quantic/cypress/e2e/default-1/sort/sort.cypress.ts

```diff
@@ -104,6 +152,64 @@ describe('quantic-sort', () => {
           });
       });
 
+      describe('when passing custom options as slots to the component', () => {
+        it('should display the custom options properly', () => {
+          const routeMatcherToUse =
+            param.useCase === useCaseEnum.insight
+              ? routeMatchers.insight
+              : routeMatchers.search;
+          cy.intercept('POST', routeMatcherToUse).as('searchRequest');
+          visitSort({useCase: param.useCase}, {withCustomSortOptions: true});
+
+          cy.wait('@searchRequest').then((interception) => {
+            Expect.sortCriteriaInSearchRequest(
+              interception.request.body,
+              defaultCustomOptionValue
+            );
+          });
+          Expect.displaySortDropdown(true);
+          Expect.labelContains('Sort by');
+
+          Actions.openDropdown();
+
+          Expect.sortOptionsEqual(customSortOptions);
+        });
+
+        customSortOptionsValues
+          .filter((value) => value !== defaultCustomOptionValue)
+          .forEach((value) => {
+            describe('when selecting a custom option', () => {
+              it(`should update the shown selected option to ${value}`, () => {
+                visitSort(
+                  {useCase: param.useCase},
+                  {withCustomSortOptions: true}
+                );
+                Actions.selectOption(value);
+
+                Expect.selectedOption(value);
+                Expect.completeSearchRequest(
+                  'resultsSort',
+                  param.useCase,
+                  (body) => Expect.sortCriteriaInSearchRequest(body, value)
+                );
+                Expect.logSortResults(value);
+              });
+            });
+          });
+
+        describe('when the custom option passed has an invalid property', () => {
```

Comment:
See this is exactly an example of what should be a unit test.

We can test this entire thing with unit tests.

Don't remove it though but this is the thing we need to be more careful on in the future.

Reply:
yeah I agree, I will leave it for now, but going forward I'll be more careful about those, especially since we had the whole discussion with the team last week